src/detectors/ScanningStrategyDetector.ts

```diff
@@ -130,7 +130,8 @@ export class ScanningStrategyDetector implements IDetector<string, ScanningStrat
         throw error;
       }
     } else if (remoteService.serviceType === ServiceType.gitlab) {
-      const { owner, repoName } = GitServiceUtils.parseUrl(remoteService.remoteUrl);
+      const { owner, repoName, host } = GitServiceUtils.parseUrl(remoteService.remoteUrl);
+      this.gitLabService.setClient(host, this.argumentsProvider.auth);
```

Comment:
This is the most important part.

It's necessary to set detected `host` from origin if the scanning strategy was determined as remote.